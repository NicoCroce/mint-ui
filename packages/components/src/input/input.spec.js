import * as React from "react";
import { fireEvent, render, cleanup } from "@testing-library/react";
import Input from "./";

export const TestInput = () => {
	const [value, setValue] = React.useState("");
	const handleChange = (val) => {
		setValue(val);
	};
	return (
		<Input
			type="text"
			placeholder="Test input"
			value={value}
			onChange={handleChange}
		/>
	);
};

afterEach(cleanup);

describe("Input tests", () => {
	test("renders correcty", () => {
		const { asFragment } = render(<TestInput />);
		expect(asFragment()).toMatchSnapshot();
	});
	test("Handles text change", () => {
		const { getByLabelText } = render(<TestInput />);
		const input = getByLabelText("tm-input");
		fireEvent.change(input, { target: { value: "New value" } });
		expect(input.value).toBe("New value");
	});
});
