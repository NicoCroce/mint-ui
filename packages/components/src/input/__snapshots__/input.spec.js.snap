// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests renders correcty 1`] = `
<DocumentFragment>
  <div
    class="input__StyledWrapper-sc-1afloio-0 bQextU"
  >
    <div
      class="input__StyledLabel-sc-1afloio-1 kZzwwB"
    >
      <label />
    </div>
    <input
      aria-label="tm-input"
      class="input__StyledInput-sc-1afloio-2 jEqpru"
      placeholder="Test input"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
